### DISCUSSION TITLE
H5 pagView 初始化报错

### DISCUSSION THREAD

**Discussion by ChuckDog** [Tencent/libpag#1636] [Q&A] [answered]
libpag 版本 ^4.2.84，H5 报错

报错位置
src/core/video-reader.ts
226 |    await this.videoEl?.play();

Unhandled Rejection (NotAllowedError): The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.

部分代码
const initPag = async () => {
    const files = pagFiles[0];
    const canvas = canvasRef.current;
    if (PAG && files && canvas) {
      const { PAGView } = PAG;
      const currentSrc = files.current;
      canvas.width = currentSrc.width();
      canvas.height = currentSrc.height();
      const pv = await PAGView.init(currentSrc, canvas); // 这里报错
      if (pv && !pagViewRef.current) {
        pagViewRef.current = pv;
        pv.setRepeatCount(0);
        await pv.play();
        dispatchProgress(100);
        dispatchVisible(false, 500);
      }
    }
  };
  useEffect(() => {
    initPag();
  }, [PAG, pagFiles, canvasRef.current]);

web 和模拟器功能运行正常，不容易定位问题。

**Comment by ChuckDog** [comment]
目前排除加载解析问题，主要怀疑是 pag 文件在手机端的兼容性。替换 pag 文件之后可以正常显示。

**Comment by zhalice2011** [comment, marked as answer]
你好, 方便提供上述两个PAG文件吗?

使用 BMP 预合成导出的 PAG 动效文件，在 Web 环境渲染是需要依赖 Video 标签进行视频解码的。

而 iOS 省电模式、部分 Android 设备以及微信浏览器中，存在“用户与页面交互之后才可以使用 Video 标签进行视频播放”的规则限制。没有点击按钮就播放的话会报错: Uncaught (in promise) DOMException: play() can only be initiated by a user gesture.

所以，如果在以上视频解码行为被限制的平台上使用 BMP 预合成导出的 PAG 动效文件时，需要在用户交互的调用中播放。或者让将 PAG 动效文件使用矢量导出。

具体兼容性描述可参考这个文档：https://github.com/Tencent/libpag/blob/main/web/doc/compatibility.md

**Reply by ChuckDog** [reply]
感谢你的回复，下面是 pag 文件。已经考虑替换文件处理。

[pag.zip](https://github.com/Tencent/libpag/files/12374732/pag.zip)
